### PR TITLE
p2p: reduce log severity

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

This PR aims to reduce the amount of `Logger.Error(..)` calls. Many of these calls are benign and do not need any intervention. 

Went from: 
```
node1    | E[2020-09-08|14:32:48.407] Connection failed @ recvRoutine (reading byte) module=p2p peer=af8747a81383f40583ae8790d2cc1f92cc7e4a35@192.167.10.4:26656 conn=MConn{192.167.10.4:26656} err="read tcp 192.167.10.3:48614->192.167.10.4:26656: read: connection reset by peer"
node1    | E[2020-09-08|14:32:48.407] Stopping peer for error                      module=p2p peer="Peer{MConn{192.167.10.4:26656} af8747a81383f40583ae8790d2cc1f92cc7e4a35 out}" err="read tcp 192.167.10.3:48614->192.167.10.4:26656: read: connection reset by peer"
node1    | E[2020-09-08|14:32:48.407] Error while stopping peer                    module=p2p peer=af8747a81383f40583ae8790d2cc1f92cc7e4a35@192.167.10.4:26656 err="already stopped"
node1    | E[2020-09-08|14:32:48.408] MConnection flush failed                     module=p2p peer=af8747a81383f40583ae8790d2cc1f92cc7e4a35@192.167.10.4:26656 err="write tcp 192.167.10.3:48614->192.167.10.4:26656: use of closed network connection"
```
To: 
```
node1    | E[2020-09-08|14:42:54.023] Stopping peer for error                      module=p2p peer="Peer{MConn{192.167.10.5:37844} e3d01d1795464a356227d0cba6567d6b94381e55 in}" err=EOF
```

Closes: #4937

### CODE REVIEW

File: p2p/switch.go [hunk 1]
```diff
@@ -246,7 +246,7 @@ func (sw *Switch) OnStop() {
 	sw.Logger.Debug("Switch: Stopping reactors")
 	for _, reactor := range sw.reactors {
 		if err := reactor.Stop(); err != nil {
-			sw.Logger.Error("error while stopped reactor", "reactor", reactor, "error", err)
+			sw.Logger.Debug("Error while stopped reactor", "reactor", reactor, "error", err)
```
Comment (marbar3778) [inline p2p/switch.go:249]:
```suggestion
			sw.Logger.Debug("Error while stopping reactor", "reactor", reactor, "error", err)
```

**Review by erikgrinaker — COMMENTED** [review]
I think this is too zealous, and many of these should remain errors. A good place to start would be to run a testnet and stop/start a few nodes, then see which error messages are emitted and silence the ones that are too noisy or misleading.

**Comment by marbar3778** [comment]
> I think this is too zealous, and many of these should remain errors. A good place to start would be to run a testnet and stop/start a few nodes, then see which error messages are emitted and silence the ones that are too noisy or misleading.

I was coming from the perspective of a node operator on what information I would like to see. It is a bit overzealous, will reduce it.

File: p2p/conn/connection.go [hunk 2]
```diff
@@ -646,7 +646,7 @@ FOR_LOOP:
 			}
 		default:
 			err := fmt.Errorf("unknown message type %v", reflect.TypeOf(packet))
-			c.Logger.Error("Connection failed @ recvRoutine", "conn", c, "err", err)
+			c.Logger.Debug("Connection failed @ recvRoutine", "conn", c, "err", err)
```
Comment (erikgrinaker) [inline p2p/conn/connection.go:649]: This should probably remain an error, since we received an unknown message type.

**Review by erikgrinaker — CHANGES_REQUESTED** [review]
Minor comment, otherwise looks good.

**Review by erikgrinaker — APPROVED** [review]

**Pull request merged**